File: database/dummy_data.py
from orders import Order
from enums import AssetClass, Side, Type

# Inject dummy data into the database
def inject_dummy_data(orders, database):
    
    database.execute("INSERT INTO cash (balance) VALUES (%s)", (423957.23,))
    
    # Bonds
    orders.send(Order('US1Y', AssetClass.BOND.value, Side.BUY.value, Type.MARKET.value, 4500, 4.9342))
    orders.send(Order('US30Y', AssetClass.BOND.value, Side.BUY.value, Type.MARKET.value, 195, 101.7422))
    orders.send(Order('US10Y', AssetClass.BOND.value, Side.BUY.value, Type.MARKET.value, 400, 99.8828))
    orders.send(Order('US5Y', AssetClass.BOND.value, Side.BUY.value, Type.MARKET.value, 255, 97.8423))
    orders.send(Order('US1Y', AssetClass.BOND.value, Side.SELL.value, Type.LIMIT.value, 10, 4.5125))
    orders.send(Order('US10Y', AssetClass.BOND.value, Side.SELL.value, Type.LIMIT.value, 200, 101.7828))
    orders.send(Order('US30Y', AssetClass.BOND.value, Side.SELL.value, Type.LIMIT.value, 10, 103.6422))
    orders.send(Order('US5Y', AssetClass.BOND.value, Side.SELL.value, Type.LIMIT.value, 45, 99.7555))
    
    # ETFs
    orders.send(Order('SPY', AssetClass.ETF.value, Side.BUY.value, Type.LIMIT.value, 50, 543.45))
    orders.send(Order('SPY', AssetClass.ETF.value, Side.BUY.value, Type.MARKET.value, 30, 543.45))
    orders.send(Order('QQQ', AssetClass.ETF.value, Side.BUY.value, Type.LIMIT.value, 100, 450.12))
    orders.send(Order('VTI', AssetClass.ETF.value, Side.BUY.value, Type.LIMIT.value, 75, 260.12))
    orders.send(Order('VOO', AssetClass.ETF.value, Side.BUY.value, Type.MARKET.value, 25, 491.06))
    orders.send(Order('SPY', AssetClass.ETF.value, Side.BUY.value, Type.LIMIT.value, 75, 530.12))
    orders.send(Order('VTI', AssetClass.ETF.value, Side.BUY.value, Type.MARKET.value, 95, 269.12))
    orders.send(Order('QQQ', AssetClass.ETF.value, Side.BUY.value, Type.MARKET.value, 50, 448.17))
    orders.send(Order('VOO', AssetClass.ETF.value, Side.BUY.value, Type.LIMIT.value, 50, 490.12))
    
    # Crypto
    orders.send(Order('BTC', AssetClass.CRYPTO.value, Side.BUY.value, Type.MARKET.value, 0.5445, 10234.13))
    orders.send(Order('ETH', AssetClass.CRYPTO.value, Side.BUY.value, Type.MARKET.value, 13.2323, 3012.50))
    orders.send(Order('BNB', AssetClass.CRYPTO.value, Side.BUY.value, Type.MARKET.value, 58.34, 547.73))
    orders.send(Order('BTC', AssetClass.CRYPTO.value, Side.SELL.value, Type.LIMIT.value, 0.2345, 66545.34))
    orders.send(Order('SOL', AssetClass.CRYPTO.value, Side.BUY.value, Type.MARKET.value, 46.45, 129.40))
    orders.send(Order('USDT', AssetClass.CRYPTO.value, Side.BUY.value, Type.MARKET.value, 10000, 1))
    orders.send(Order('USDC', AssetClass.CRYPTO.value, Side.BUY.value, Type.MARKET.value, 10000, 1))
    orders.send(Order('BNB', AssetClass.CRYPTO.value, Side.SELL.value, Type.LIMIT.value, 21.34, 589.21))
    orders.send(Order('ETH', AssetClass.CRYPTO.value, Side.SELL.value, Type.LIMIT.value, 6.1234, 3112.50))
    
    # Stocks
    orders.send(Order('MS', AssetClass.STOCK.value, Side.BUY.value, Type.MARKET.value, 89, 105.89))
    orders.send(Order('TSLA', AssetClass.STOCK.value, Side.BUY.value, Type.LIMIT.value, 75, 210.12))
    orders.send(Order('AAPL', AssetClass.STOCK.value, Side.BUY.value, Type.MARKET.value, 150, 205.83))
    orders.send(Order('AMZN', AssetClass.STOCK.value, Side.BUY.value, Type.LIMIT.value, 100, 170.12))
    orders.send(Order('META', AssetClass.STOCK.value, Side.BUY.value, Type.LIMIT.value, 5, 469.97))
    orders.send(Order('MSFT', AssetClass.STOCK.value, Side.BUY.value, Type.LIMIT.value, 50, 426.23))
    orders.send(Order('GOOG', AssetClass.STOCK.value, Side.BUY.value, Type.LIMIT.value, 20, 298.12))
    orders.send(Order('TSLA', AssetClass.STOCK.value, Side.BUY.value, Type.LIMIT.value, 25, 202.32))
    orders.send(Order('MS', AssetClass.STOCK.value, Side.BUY.value, Type.LIMIT.value, 120, 90.12))
    orders.send(Order('META', AssetClass.STOCK.value, Side.BUY.value, Type.LIMIT.value, 50, 469.97))
    orders.send(Order('MSFT', AssetClass.STOCK.value, Side.BUY.value, Type.LIMIT.value, 80, 380.56))
    orders.send(Order('NVDA', AssetClass.STOCK.value, Side.BUY.value, Type.MARKET.value, 45, 101.94))
    orders.send(Order('GOOG', AssetClass.STOCK.value, Side.BUY.value, Type.LIMIT.value, 75, 299.72))
    orders.send(Order('AAPL', AssetClass.STOCK.value, Side.SELL.value, Type.LIMIT.value, 75, 234.12))
    
    # Add performance data
    values = [("2023-09-01",  345689.56),
              ("2023-10-01",  365464.90),
              ("2023-11-01",  354685.47),
              ("2023-12-01",  376855.23),
              ("2024-01-01",  366855.68),
              ("2024-02-01",  395000.67),
              ("2024-03-01",  406340.93),
              ("2024-04-01",  422674.58),
              ("2024-05-01",  414468.92),
              ("2024-06-01",  425152.71),
              ("2024-07-01",  433675.45)]
    
    database.executemany("INSERT INTO performance (date, total_value) VALUES (%s, %s)", values)
    

File: web/scripts/dummy_data.js
import { AssetClass, Side, Status, Type } from './enums.js';

const dummy_positions = [
    ['AAPL', 'Stock', 100, 200],
    ['MSFT', 'Stock', 70, 250],
    ['NVDA', 'Stock', 80, 250],
    ['META', 'Stock', 60, 250],
    ['GOOG', 'Stock', 40, 250],
    ['AMZN', 'Stock', 50, 250],
    ['BTC', 'Crypto', 0.2445, 15468.39],
    ['ETH', 'Crypto', 1.2323, 3209.80],
    ['USDT', 'Crypto', 10000, 1],
    ['BNB', 'Crypto', 20.34, 576.83],
    ['SOL', 'Crypto', 12.13, 185.92],
    ['USDC', 'Crypto', 10000, 0.9999],
    ['SPY', 'ETF', 80, 250],
    ['VOO', 'ETF', 80, 250],
    ['VTI', 'ETF', 80, 250],
    ['QQQ', 'ETF', 80, 250],
    ['US1Y', 'Bond', 1000, 250],
    ['US5Y', 'Bond', 150, 250],
    ['US10Y', 'Bond', 150, 250],
    ['US30Y', 'Bond', 150, 250]
]

const dummy_orders = [
    [1, 'MSFT', Side.Buy, Type.Market, AssetClass.Stock, 100, 100, 'l-kesJn07NNNYzeaGnWJ', Status.Filled, 100],
    [2, 'AAPL', Side.Sell, Type.Market, AssetClass.Stock, 50, 100, 'tu0lj408rhA3l8t2XoBz', Status.PartiallyFilled, 50],
    [3, 'MSFT', Side.Buy, Type.Market, AssetClass.Stock, 100, 100, 'Cw9E7m80aHGEiTnBv6hN', Status.Rejected, 100],
    [4, 'AAPL', Side.Sell, Type.Market, AssetClass.Stock, 50, 100, 'g1a-OP5xjU2eUxr6xcGE', Status.New, 50],
    [5, 'MSFT', Side.Buy, Type.Market, AssetClass.Stock, 100, 100, '42sVu2XbvU128vyzKQ3V', Status.Filled, 100],
    [6, 'AAPL', Side.Sell, Type.Market, AssetClass.Stock, 50, 100, 'WKOWV1q6ZrMA6jrs6YT7', Status.Filled, 50],
    [7, 'MSFT', Side.Buy, Type.Market, AssetClass.Stock, 100, 100, 'wCMi6LmT8I8EO1hfdwfh', Status.PartiallyFilled, 100],
    [8, 'AAPL', Side.Sell, Type.Market, AssetClass.Stock, 50, 100, 'FXv7eGXifqlPzmW6VCWf', Status.Filled, 50],
    [9, 'MSFT', Side.Buy, Type.Market, AssetClass.Stock, 100, 100, 'd2SHxFRpNTX5UlUIYPAK', Status.New, 100],
    [10, 'AAPL', Side.Sell, Type.Market, AssetClass.Stock, 50, 100, 'BrgwAuLmHgbINVFzD3Ri', Status.Rejected, 50],
    [11, 'MSFT', Side.Buy, Type.Market, AssetClass.Stock, 100, 100, 'LcG1RuwwTK8GvLERsMnJ', Status.New, 100],
    [12, 'AAPL', Side.Sell, Type.Market, AssetClass.Stock, 50, 100, 'C3as2jy8Z3yV1n8l5HgF', Status.Filled, 50]]

// Generate performance data programmatically
// Last date is 1 month ago from current date, each previous date goes 1 month back
// Performance values are based on offsets from the current portfolio value
function generatePerformanceData(portfolioValue) {
    const currentDate = new Date();
    const lastMonth = new Date(currentDate.getFullYear(), currentDate.getMonth() - 1, currentDate.getDate());

    // Percentage offsets calculated so that when portfolioValue = 460000, 
    // the results approximate the dummy_performance values
    const percentageOffsets = [
        -0.2485,
        -0.2055,
        -0.2289,
        -0.1808,
        -0.2025,
        -0.1413,
        -0.1166,
        -0.0812,
        -0.0989,
        -0.0758,
        -0.0572
    ];

    const performanceData = [];

    for (let i = 0; i < 11; i++) {
        // Start from 10 months before last month and work forward
        const date = new Date(lastMonth.getFullYear(), lastMonth.getMonth() - (10 - i), currentDate.getDate());
        const dateStr = date.toISOString().slice(0, 10); // Format as YYYY-MM-DD

        // Calculate value based on offset from current portfolio value
        const historicalValue = portfolioValue * (1 + percentageOffsets[i]);
        performanceData.push([dateStr, historicalValue]);
    }

    return performanceData;
}

const dummy_cash = 113957.23;

export { dummy_positions, dummy_orders, dummy_cash, generatePerformanceData };